next.data.mjs

```diff
@@ -1,24 +1,24 @@
 import * as preBuild from './scripts/next-data/generatePreBuildFiles.mjs';
 
 import getLocalisationData from './scripts/next-data/getLocalisationData.mjs';
-import getNodeVersionData from './scripts/next-data/getNodeVersionData.mjs';
 import getBlogData from './scripts/next-data/getBlogData.mjs';
+import generateNodeReleasesData from './scripts/next-data/generateNodeReleasesData.mjs';
 
 const cachedBlogData = getBlogData();
 
+await generateNodeReleasesData();
```

Comment:
No need to add an `await` here, we don't want to block the main context.

Reply:
If `await` is removed, main process won't wait until the pending Promise is resolved.
Practically, `generateNodeReleasesData` won't outlive the main process I think. I tried this case by adding a long `setTimeout`.

I am in favour of keeping the `await`. Not to mention that both `generateBlogYearPages` and `generateWebsiteFeeds` is blocking too.

https://www.stefanjudis.com/today-i-learned/top-level-promise-handling-in-node-js/#pending-promises-don't-keep-the-node-event-loop-alive

Reply:
> I am in favour of keeping the await. Not to mention that both generateBlogYearPages and generateWebsiteFeeds is blocking too.

Where they're blocking? I don't see them blocking the main thread :)

They block each individual route. And their promises get fulfilled only once, then reused on every following route.

Since this data is never used during the server-side generation, we shouldn't block the main thread with this. Not to mention this process should be fast enough.

Also, I'm usually not a fan of a blocking approach; the whole point of Node.js is to be a non-blocking main thread.

https://nodejs.dev/en/learn/overview-of-blocking-vs-non-blocking/

Reply:
I misread the code :)
The final `.then()` returns a Promise.
Yes, you're right. It's not blocking.
```js
export const generateBlogYearPages = cachedBlogData =>
  cachedBlogData('', true)
    .then(({ blogData }) => blogData.posts.map(p => p.date.getFullYear()))
    .then(data => [...new Set(data)])
    .then(data => data.forEach(createBlogYearFile));
```